pkg/account/delete/client.go

```diff
@@ -0,0 +1,176 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package delete
+
+import (
+	"errors"
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/clients/accounts"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"golang.org/x/net/context"
+	"io"
+	"net/http"
+)
+
+// Client for deleting resources from the Account Management API
+type Client interface {
+	DeleteUser(ctx context.Context, email string) error
+	DeleteGroup(ctx context.Context, name string) error
+	DeleteAccountPolicy(ctx context.Context, name string) error
+	DeleteEnvironmentPolicy(ctx context.Context, environment, name string) error
+	GetAccountUUID() string
```

Comment:
Ah, so that you can mock it. I'd still think about removing this method, as it's only necessary for the AccountResources method (or did I overlook something?)

Reply:
After a fourth look, I would even argue that the unit test is in that version not really necessary (delete_test.go w/ custom client). We could just use an http-server, like in the client_test, to verify if the resources have been deleted. 
Then we have an actual full package test instead of testing some independent parts. And the Client interface could go.

Reply:
as mentioned in the notes, the interface, especially in it's latest form is very much up for discussion for me. Don't like it 100% either. 

Why is there an "accountUUID" getter? Because the latest refactor moves the uuid into the client to ensure you don't even try to use a client with different UUIDs - authentication is bound to one account.
The getter currently just exists for logging though... 

Why the client interface at all? It allows testing the logical behavior of delete (continue on errors) independently of the API details. Notably it allowed to write and test that, before getting down to the details of the API. Could be removed again, but then the methods would stay as functions anyway and the small delete logic test get's more complex to write and read IMO...